### PR TITLE
17796 Allow firm registration for single name user/account

### PR DESCRIPTION
**Pull request by chenhongjing**
*Issue #:* /bcgov/entity#17796

*Description of changes:*
- the first name field is optional for completing party if the party is `person`
- the last name field should be non-empty for completing party if the party is `person`
- fixed the example data

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the business-schemas license (Apache 2.0).

### CODE REVIEW

File: src/registry_schemas/schemas/party.json [hunk 1]
```diff
@@ -119,7 +120,48 @@
...
+                ]
+            },
+            "then": {
+                "properties": {
+                  "officer": {
+                    "type": "object",
+                    "properties": {
+                      "firstName": { "minLength": 1 }
+                    },
+                    "required": ["firstName"]
```
Comment (argush3) [inline src/registry_schemas/schemas/party.json:161]: i thought `lastName` should be the required field when the role is "Completing Party"

File: src/registry_schemas/example_data/schema_data.py [hunk 2]
```diff
@@ -958,8 +958,8 @@
         {
             'officer': {
                 'id': 2,
-                'firstName': '',
-                'lastName': '',
+                'firstName': 'Peter',
```
Comment (thorwolpert) [inline src/registry_schemas/example_data/schema_data.py:961]: You shouldn't be able to have name fields and organizationName

File: src/registry_schemas/schemas/party.json [hunk 3]
```diff
@@ -119,7 +120,48 @@
                 "officer",
                 "roles",
                 "mailingAddress"
-            ]
+            ],
+            "if": {
```
Comment (thorwolpert) [inline src/registry_schemas/schemas/party.json:124]: In the schema, we can just make the lastName mandatory, and everything else optional.
As eventually that will come in to affect for all natural persons.

**Review by thorwolpert — COMMENTED** [review]
The names examples break some fundamental rules.
The natural person should only require a lastName, leave out the complexity.

**Review by thorwolpert — APPROVED** [review]
Thanks for the changes

**Review by argush3 — APPROVED** [review]

**Pull request merged**